### PR TITLE
improve [`match_same_arms`] messages, enable rustfix test

### PR DESCRIPTION
**Pull request by J-ZhengLi**
closes: #9251 

don't worry about the commit size, most of them are generated

---

changelog: improve [`match_same_arms`] lint messages

### CODE REVIEW

File: clippy_lints/src/matches/match_same_arms.rs [hunk 1]
```diff
@@ -144,23 +146,36 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, arms: &'tcx [Arm<'_>]) {
...
                     )
-                    .help("or try changing either arm body")
-                    .span_note(move_arm.span, "other arm here");
+                    .span_suggestion(
+                        adjusted_arm_span(cx, move_arm.span),
+                        "and remove this obsolete arm",
+                        "",
+                        appl,
+                    )
+                    .help("or try changing either arm body");
```
Comment (J-ZhengLi) [inline clippy_lints/src/matches/match_same_arms.rs:164]: is there a way to ensure the help message is printed after the suggestions?

Comment (blyxyas) [reply clippy_lints/src/matches/match_same_arms.rs:164]: Seems like the suggestions are printed after any non-suggestion messages. You can try rephrasing the help messages taking that into account

**Review by blyxyas — CHANGES_REQUESTED** [review]
LGTM, thanks! :heart: 
Could you rephrase the help messages?

**Comment by blyxyas** [comment]
I've manually done it (I was bored and wanted to take this through the finish line, being that such an easy-to-solve issue was blocking it)

@bors r+

**Comment by J-ZhengLi** [comment]
> I've manually done it (I was bored and wanted to take this through the finish line, being that such an easy-to-solve issue was blocking it)
> 
> @bors r+

Thanks! that'll do 

I was too tired back then, and was trying to think of some better help message but ended up fall asleep 🤦‍♂️ 

As of the failed test, I've encounter the exactly problem in the past with Alexendoo iirc, it was strange, ended up re-open another PR to solve it, I'll try change the commit message to see if that goes away

**Comment by y21** [comment]
I think this happens because of the "requested changes" review. You need to dismiss that (or blyxyas needs to approve it) before bors will merge it.

**Comment by J-ZhengLi** [comment]
> I think this happens because of the "requested changes" review. You need to dismiss that (or blyxyas needs to approve it) before bors will merge it. Like here: [#12107 (comment)](https://github.com/rust-lang/rust-clippy/pull/12107#issuecomment-2106303319)

did a quick google search and it's telling me that you need write access to dismiss it, well I guess I'll have to rely on blyxyas then 😸

**Review by blyxyas — APPROVED** [review]
